src/ui/components/instantiate/Wizard.tsx

```diff
@@ -2,36 +2,27 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import React from 'react';
-import { FormField, getValidation } from '../form/FormField';
-import { AccountSelect } from '../account/AccountSelect';
 import { Step1 } from './Step1';
 import { Step2 } from './Step2';
 import { Step3 } from './Step3';
 import { Stepper } from './Stepper';
-import { CONTRACT_FILE, DEPLOYMENT_INFO, FINALIZE, useInstantiate } from 'ui/contexts';
 
-const steps = ['Contract Bundle', 'Deployment Info', 'Confirmation'];
+const steps = [
+  { name: 'Contract Bundle', index: 1 },
+  { name: 'Deployment Info', index: 2 },
+  { name: 'Confirmation', index: 3 },
+];
 
 export function Wizard() {
-  const {
-    accountId,
-    step: [step],
-  } = useInstantiate();
-
   return (
     <div className="grid md:grid-cols-12 gap-5 m-1">
       <main className="md:col-span-9 p-4">
-        {step < 2 && (
-          <FormField className="mb-8" id="accountId" label="Account" {...getValidation(accountId)}>
-            <AccountSelect id="accountId" className="mb-2" {...accountId} />
-          </FormField>
-        )}
-        {step === CONTRACT_FILE && <Step1 />}
-        {step === DEPLOYMENT_INFO && <Step2 />}
-        {step === FINALIZE && <Step3 />}
+        <Step1 />
```

Comment:
Any reason not to selectively render each step based on the current index? Rendering all three seems to do some unneeded work.

Reply:
this is intentional. 
- when you return `null` from within the component it will not unmount, it will just not render. This way it will keep it's state when you click back and forward through the wizard
- when you do the check in the parent component, the components are unmounted and they loose state.

the "work" of preserving form state between steps needs to happen anyway, one way of the other.
this approach has measurable better performance than holding form state in context, for example.